sandbox/lua/decoders/bind_query_log.lua

```diff
@@ -0,0 +1,274 @@
+      C = "no DNSSEC validation check",
+      T = "TCP used",
+}
+
+--Create a capture group that uses the table above to match any one or more of '+-EDCST'
+--that are present and add them to a new table called QueryFlags: 
+query_flags = l.Cg( l.Ct((l.S"+-EDCST" / t)^1), "QueryFlags" )
+
+--[[Hostname and domain name patterns
+
+Hostnames and domain names are broken up into fragments that are called 
+"labels", which are the parts between the dots.
+
+Source: https://en.wikipedia.org/wiki/Hostname#Restrictions_on_valid_host_names
+
+For instance, webserver.company.com has the labels "webserver", "company" and 
+"com"
+
+The pattern below uses the upper, lower and digit shortcuts from the main LPEG 
+library and combines them with the hyphen (-) character to match anything that's
+part of a valid hostname label. The ^1 means match one or more instances of it.
+--]]
+local hostname_fragment = (l.upper + l.lower + l.digit +  "-")^1
+
+--The pattern below matches one or more hostname_fragments, followed by a . 
+--and followed by one more hostname_fragment, indicating the end of a complete
+--hostname. The open and close parens and colon are to match the decorations
+--BIND puts around the name: 
+-- (webserver.company.com):
+local enclosed_query = "(" * l.Cg((hostname_fragment * ".")^1 * hostname_fragment, "FullQuery") * "):"
+
+--The ^-1 means match at most 1 instance of the pattern. We want this so that we 
+--can match the first part of a hostname and leave the rest for the l.Cg((hostname_fragment...
+--capture group to match into the QueryDomain table entry.
+--In webserver.company.com, `(hostname_fragment * ".")^-1` matches webserver.
+--and `l.Cg((hostname_fragment...` matches company.com
+local query = l.Cg((hostname_fragment)^-1, "QueryName") * "." * l.Cg((hostname_fragment * ".")^1 * hostname_fragment, "QueryDomain")
+
+--Use all of the previously defined patterns to build a grammar:
+local bind_query = timestamp * space * queries_literal * space * info_literal * space * client_literal * space * client_ip * space * enclosed_query * space * query_literal * space * query * space * dns_record_class * space * dns_record_type * space * query_flags * space * server_responding_ip
+
+-- To use the above grammar on http://lpeg.trink.com/, paste in everything above this line and just the single line below (uncomment it too):
+--grammar = l.Ct(bind_query)
+
+local msg = {
+  --This value is read in from the 'msg_type' config option in the TOML:
+  Type        = msg_type,
+  Payload     = nil,
+  Severity    = 'info',
+  Fields      = {},
+}
+
+function process_message ()
+
+  --Create a local variable for the log line from the Payload of the incoming Heka message.
+  --The LogstreamerInput that generates the messages automatically puts the log line in the Payload:
+  local query_log_line = read_message("Payload")
+  --Use the bind_query grammar defined farther above to create a local grammar object:
+  local grammar = l.Ct(bind_query)
```

Comment:
Ideally this grammar would be made into a module so it could be re-used.  To  be used with Heka it would need to go into the heka branch of the lua_sandbox